desc/compute/_stability.py

```diff
@@ -231,3 +231,513 @@ def _magnetic_well(params, transforms, profiles, data, **kwargs):
+        "cvdrift",
+        "cvdrift0",
+        "|B|",
+        "B^zeta",
+        "p_r",
+        "phi",
+        "iota",
+        "iota_r",
+        "psi",
+        "psi_r",
+        "rho",
+    ],
+)
+def _ideal_ballooning_gamma1(params, transforms, profiles, data, *kwargs):
+    """
+    Ideal-ballooning growth rate finder.
+
+    This function uses a finite-difference method
+    to calculate the maximum growth rate against the
+    infinite-n ideal ballooning mode. The equation being solved is
+
+    d/dζ(g dX/dζ) + c X = lam f X, g, f > 0,
+
+    where
+
+    𝛋 = 𝐛 ⋅∇ 𝐛
+    g = a_N^3 * B_N * (b ⋅∇ζ) * |∇α|², / B,
+    c = a_N/B_N * (1/ b ⋅∇ζ) * dψ/dρ * dp/dψ * (b × 𝛋) ⋅|∇α|/ B**2,
+    f = a_N * B_N^3 *|∇α|² / B^3 * (1/ b ⋅∇ζ) ,
+
+    are needed along a field line to solve the ballooning equation once.
+
+    To obtain the parameters g, c, and f, we need a set of parameters
+    provided in the list ``data`` above. Here's a description of
+    these parameters:
+
+    - a: minor radius of the device
+    - g^aa: |grad alpha|^2, field line bending term
+    - g^ra: (grad alpha dot grad rho) integrated local shear
+    - g^rr: |grad rho|^2 flux expansion term
+    - cvdrift: geometric factor of the curvature drift
+    - cvdrift0: geoetric factor of curvature drift 2
+    - |B|: magnitude of the magnetic field
+    - B^zeta: inverse of the jacobian
+    - p_r: dp/drho, pressure gradient
+    - phi: coordinate describing the position in the toroidal angle
+    along a field line
+    """
+    rho = data["rho"]
+
+    psi_b = params["Psi"] / (2 * jnp.pi)
+    a_N = data["a"]
+    B_N = 2 * psi_b / a_N**2
+
+    N_zeta0 = int(11)
+    # up-down symmetric equilibria only
+    zeta0 = jnp.linspace(-jnp.pi / 2, jnp.pi / 2, N_zeta0)
+
+    iota = data["iota"]
+    shat = -rho / iota * data["iota_r"]
+    psi = data["psi"]
+    sign_psi = jnp.sign(psi[-1])
+    sign_iota = jnp.sign(iota[-1])
+
+    phi = data["phi"]
+
+    N_alpha = int(8)
+    N_zeta = int(len(phi) / N_alpha)
+
+    B = jnp.reshape(data["|B|"], (N_alpha, 1, N_zeta))
+    gradpar = jnp.reshape(data["B^zeta"] / data["|B|"], (N_alpha, 1, N_zeta))
+    dpdpsi = jnp.mean(mu_0 * data["p_r"] / data["psi_r"])
+
+    gds2 = jnp.reshape(
+        rho**2
+        * (
+            data["g^aa"][None, :]
+            - 2 * sign_iota * shat / rho * zeta0[:, None] * data["g^ra"][None, :]
+            + zeta0[:, None] ** 2 * (shat / rho) ** 2 * data["g^rr"][None, :]
+        ),
+        (N_alpha, N_zeta0, N_zeta),
+    )
+
+    f = a_N**3 * B_N * gds2 / B**3 * 1 / gradpar
+    g = a_N**3 * B_N * gds2 / B * gradpar
+    g_half = (g[:, :, 1:] + g[:, :, :-1]) / 2
+    c = (
+        1
+        * a_N**3
+        * B_N
+        * jnp.reshape(
+            2
+            / data["B^zeta"][None, :]
+            * sign_psi
+            * rho**2
+            * dpdpsi
+            * (
+                data["cvdrift"][None, :]
+                - shat / rho * zeta0[:, None] * data["cvdrift0"][None, :]
```

Comment:
Push the factor of $2 \rho^2/\rho = 2\rho$ into `cvdrift0` to be consistent with the term as defined in Gamma_c https://github.com/PlasmaControl/DESC/pull/1042#discussion_r1638688019